handlers/analytics.go

```diff
@@ -35,34 +37,60 @@ type AnalyticsLog struct {
 }
 
 type AnalyticsLogEvent struct {
-	Type           string `json:"type"`
-	Timestamp      int64  `json:"timestamp"`
-	Errors         int    `json:"errors,omitempty"`
-	PlaytimeMS     int    `json:"playtime_ms,omitempty"`
-	TTFFMS         int    `json:"ttff_ms,omitempty"`
-	PreloadTimeMS  int    `json:"preload_time_ms,omitempty"`
-	AutoplayStatus string `json:"autoplay_status,omitempty"`
-	BufferMS       int    `json:"buffer_ms,omitempty"`
-	ErrorMessage   string `json:"error_message,omitempty"`
+	// Shared fields by all events
+	Type      string `json:"type"`
+	Timestamp int64  `json:"timestamp"`
+
+	// Heartbeat event
+	Errors              int    `json:"errors"`
+	AutoplayStatus      string `json:"autoplay_status"`
+	StalledCount        int    `json:"stalled_count"`
+	WaitingCount        int    `json:"waiting_count"`
+	TimeErroredMS       int    `json:"time_errored_ms"`
+	TimeStalledMS       int    `json:"time_stalled_ms"`
+	TimePlayingMS       int    `json:"time_playing_ms"`
+	TimeWaitingMS       int    `json:"time_waiting_ms"`
+	MountToPlayMS       int    `json:"mount_to_play_ms"`
+	MountToFirstFrameMS int    `json:"mount_to_first_frame_ms"`
+	PlayToFirstFrameMS  int    `json:"play_to_first_frame_ms"`
+	DurationMS          int    `json:"duration_ms"`
+	OffsetMS            int    `json:"offset_ms"`
+	PlayerHeightPX      int    `json:"player_height_px"`
+	PlayerWidthPX       int    `json:"player_width_px"`
+	VideoHeightPX       int    `json:"video_height_px"`
+	VideoWidthPX        int    `json:"video_width_px"`
+	WindowHeightPX      int    `json:"window_height_px"`
+	WindowWidthPX       int    `json:"window_width_px"`
+
+	// Error event
+	ErrorMessage string `json:"error_message"`
+	Category     string `json:"category"`
 }
 
 type AnalyticsGeo struct {
 	GeoHash     string
+	Continent   string
 	Country     string
+	CountryCode string
 	Subdivision string
 	Timezone    string
+	IP          string
 }
 
 type AnalyticsHandlersCollection struct {
 	extFetcher   analytics.IExternalDataFetcher
 	logProcessor analytics.ILogProcessor
 }
 
-func NewAnalyticsHandlersCollection(streamCache mistapiconnector.IStreamCache, lapi *api.Client, metricsURL string, host string) AnalyticsHandlersCollection {
+func NewAnalyticsHandlersCollection(streamCache mistapiconnector.IStreamCache, lapi *api.Client, kafkaBootstrapServers, kafkaUser, kafkaPassword, kafkaTopic string) (AnalyticsHandlersCollection, error) {
+	lp, err := analytics.NewLogProcessor(kafkaBootstrapServers, kafkaUser, kafkaPassword, kafkaTopic)
+	if err != nil {
+		return AnalyticsHandlersCollection{}, err
+	}
```

Comment:
**nit:** WDYT of receiving only the `analytics.ILogProcessor` as an input to this constructor, instead of the args for creating the processor? Will keep the interface much cleaner IMO and consistent with how the `lapi` is handled. 

The specific Kafka stuff would be initialized on the place that creates this instead (plus more explicit error handling) and we can deal with only the interface here.

Reply:
I like it :) Updated.